script/deploy/mainnet/012_xOGNGovernanceScript.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "./BaseMainnetScript.sol";
+import {Vm} from "forge-std/Vm.sol";
+
+import {Addresses} from "contracts/utils/Addresses.sol";
+
+import {FixedRateRewardsSourceProxy} from "contracts/upgrades/FixedRateRewardsSourceProxy.sol";
+import {ExponentialStakingProxy} from "contracts/upgrades/ExponentialStakingProxy.sol";
+import {MigratorProxy} from "contracts/upgrades/MigratorProxy.sol";
+
+import {ExponentialStaking} from "contracts/ExponentialStaking.sol";
+import {FixedRateRewardsSource} from "contracts/FixedRateRewardsSource.sol";
+import {OgvStaking} from "contracts/OgvStaking.sol";
+import {Migrator} from "contracts/Migrator.sol";
+import {Timelock} from "contracts/Timelock.sol";
+import {Governance} from "contracts/Governance.sol";
+
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/token/ERC20/extensions/ERC20Votes.sol";
+import "OpenZeppelin/openzeppelin-contracts@4.6.0/contracts/governance/TimelockController.sol";
+
+// To be extracted
+library GovFive {
+    struct GovFiveAction {
+        address receiver;
+        string fullsig;
+        bytes data;
+    }
+
+    struct GovFiveProposal {
+        string name;
+        string description;
+        GovFiveAction[] actions;
+    }
+
+    function setName(GovFiveProposal storage prop, string memory name) internal {
+        prop.name = name;
+    }
+
+    function setDescription(GovFiveProposal storage prop, string memory description) internal {
+        prop.description = description;
+    }
+
+    function action(GovFiveProposal storage prop, address receiver, string memory fullsig, bytes memory data)
+        internal
+    {
+        prop.actions.push(GovFiveAction({receiver: receiver, fullsig: fullsig, data: data}));
+    }
+
+    function execute(GovFiveProposal storage prop) internal {
+        address VM_ADDRESS = address(uint160(uint256(keccak256("hevm cheat code"))));
+        Vm vm = Vm(VM_ADDRESS);
+        for (uint256 i = 0; i < prop.actions.length; i++) {
+            GovFiveAction memory action = prop.actions[i];
+            bytes memory sig = abi.encodePacked(bytes4(keccak256(bytes(action.fullsig))));
+            vm.prank(Addresses.TIMELOCK);
+            action.receiver.call(abi.encodePacked(sig, action.data));
+        }
+    }
+}
+
+contract XOGNGovernanceScript is BaseMainnetScript {
```

Comment:
In the real world this will be the multisig calling the timelock. However this is a lower priority. First we need to make sure that main switch over has enough tests on it.